### PR TITLE
Course-subject-task

### PR DESCRIPTION
**Pull request by leminhtuan2015**

### CODE REVIEW

File: app/controllers/subjects_controller.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    
+	  @subject = Subject.find(params[:id])
+    if @subject.update_attributes subject_params
+      redirect_to subject_path(@subject)
+    else
+      render 'edit'
+    end
+  end
+
+	def destroy
```
Comment (chintk) [inline app/controllers/subjects_controller.rb]: intent

File: app/controllers/tasks_controller.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+class TasksController < ApplicationController
+  def create
+  	if !params[:name].empty?
```
Comment (chintk) [inline app/controllers/tasks_controller.rb]: intent

File: app/controllers/tasks_controller.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      @task = Task.new(task_params)
+      @task.save
+    end  
+	  redirect_to subject_path params[:subject_id]
```
Comment (chintk) [inline app/controllers/tasks_controller.rb]: intent

File: app/controllers/tasks_controller.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    @task = Task.find params[:id]
+    @task.destroy
+    redirect_to subject_path @task.subject_id
+  end
+
+  private
+  def task_params
+      params.permit(:name, :description, :subject_id)
+  end
+
```
Comment (chintk) [inline app/controllers/tasks_controller.rb]: line

File: app/views/courses/_remove_subject.html.erb [hunk 5]
```diff
@@ -0,0 +1,34 @@
+<table class="table">
+  <thead>
+    <tr>
+      <th>#</th>
+        <th>Name</th>
```
Comment (chintk) [inline app/views/courses/_remove_subject.html.erb]: intent

File: app/controllers/application_controller.rb [hunk 6]
```diff
@@ -10,4 +10,23 @@ def logged_in_user
...
+  def update_course_subject(params)
+    if params.has_key?(:course_id)
+      course_id = params[:course_id]
+    else
+      course_id = params[:id]
+    end
+    CourseSubject.where(:course_id => course_id.to_i).delete_all
+    if params.has_key?(:subjects)
+      params[:subjects].each do |s|
+        course_subject = CourseSubject.new
```
Comment (chintk) [inline app/controllers/application_controller.rb]: intent

File: app/controllers/course_subjects_controller.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    Course.find(1).containing(1)
+  end
+
+  def create
+    update_course_subject params
+    redirect_to course_path params[:course_id]
+  end
+
+  def destroy
+
```
Comment (chintk) [inline app/controllers/course_subjects_controller.rb]: line

File: app/controllers/subjects_controller.rb [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    @subject = Subject.find params[:id]
+    @task = Task.new
+  end
+
+  def update
+    if params[:name]!=""
+      @task = Task.new params.permit(:name, :description, :subject_id)
+      @task.save
+    end
+
```
Comment (chintk) [inline app/controllers/subjects_controller.rb]: line

File: app/views/courses/_remove_subject.html.erb [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+      <th>Name</th>
+      <th>Select</th>
+      <th>Included</th>
+    </tr>
+  </thead>
+  <tbody>
+    <% @subjects.each_with_index do |subject, i| %>
+      <tr>
+        <th scope="row"><%=subject.id%></th> 
+          <td>
```
Comment (chintk) [inline app/views/courses/_remove_subject.html.erb]: intent

File: app/views/courses/show.html.erb [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+      <th>Subject</th>
+      <th>Trainner</th>
+      <th>Supervisor</th>
+      <th>Update</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th scope="row"><%=@course.id%></th>
+        <td><%= link_to @course.name, course_path(@course)%>
```
Comment (chintk) [inline app/views/courses/show.html.erb]: content

File: app/views/courses/show.html.erb [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+        </td>
+        <td><%= @course.description %></td>
+        <td><%= @course.description %></td>
+        <td>
+          <%= link_to "Update", edit_course_path(@course)%>
+          <%= link_to "Delete", course_path(@course),
+                       method: :delete,
+                       data: { confirm: 'Drop this course?' }%>
+        </td>
+      </tr>
```
Comment (chintk) [inline app/views/courses/show.html.erb]: intent

File: app/views/subjects/_all_subjects.html.erb [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+      <th>#</th>
+      <th>Name</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <% @subjects.each do |subject| %>
+      <tr>
+        <th scope="row"><%=subject.id%></th>
+          <td>
```
Comment (chintk) [inline app/views/subjects/_all_subjects.html.erb]: intent

File: app/views/tasks/_tasks.html.erb [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+      <th>Name</th>
+      <th>Description</th>
+      <th>Actions</th>
+    </tr>
+  </thead>
+  <tbody>
+    <% @subject.tasks.each do |task| %>
+      <tr>
+        <th scope="row"><%=task.id%></th>
+          <td>
```
Comment (chintk) [inline app/views/tasks/_tasks.html.erb]: indent

File: config/routes.rb [hunk 14]
```diff
@@ -9,4 +9,17 @@
...
+
+  resources :subjects do
+  end
+
+  resources :course_subjects do
+  end
+
+  resources :tasks do
+  end
+
```
Comment (chintk) [inline config/routes.rb]: line

File: app/controllers/courses_controller.rb [hunk 15]
```diff
@@ -0,0 +1,53 @@
+class CoursesController < ApplicationController
+  include ApplicationHelper
+
+  def index
+    @courses = Course.all
+  end
+
+  def create	
+    @course = Course.new(course_params)
```
Comment (athuan) [inline app/controllers/courses_controller.rb]: remove ()

File: app/controllers/courses_controller.rb [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+
+  def destroy
+    @course = Course.find params[:id]
+    @course.destroy
+    redirect_to courses_path
+  end
+
+  def update
+    update_course_subject params
+    @course = Course.find(params[:id])
```
Comment (athuan) [inline app/controllers/courses_controller.rb]: remove ()

File: app/controllers/subjects_controller.rb [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+    @subject = Subject.find params[:id]
+    @task = Task.new
+  end
+
+  def new
+    @subject = Subject.new
+  end
+
+  def create
+    @subject = Subject.new(subject_params)
```
Comment (athuan) [inline app/controllers/subjects_controller.rb]: remove ()

**Pull request closed without merging**